### PR TITLE
feat(dual-alb-fargate): dual-alb-fargate support grpc application

### LINKED ISSUE #69: Add ALB gRPC support
background: https://aws.amazon.com/blogs/aws/new-application-load-balancer-support-for-end-to-end-http-2-and-grpc/

I think we need to be able to specify protocolVersion on the target group? Also we should be able to define a healthcheck using "healthyGrpcCodes", perhaps already possible? 

Thanks

### PR DESCRIPTION
**Pull request by neilkuan**
feat(dual-alb-fargate): dual-alb-fargate support grcp application
docs(readme): add grcp example code

close #69

### CODE REVIEW

File: services/python-grcp/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM python:3.7
```
Comment (pahud) [inline services/python-grcp/Dockerfile:1]: please rename `python-grcp` to `python-grpc`

File: src/dual-alb-fargate-svc.ts [hunk 2]
```diff
@@ -45,7 +45,9 @@ export class DualAlbFargateService extends BaseFargateService {
       if (t.external) {
         const exttg = new elbv2.ApplicationTargetGroup(this, `${defaultContainerName}ExtTG`, {
           protocol: elbv2.ApplicationProtocol.HTTP,
+          port: t.enablegRPC ? t.external!.port! : undefined,
           vpc: this.vpc,
+          protocolVersion: t.enablegRPC ? elbv2.ApplicationProtocolVersion.GRPC : undefined,
```
Comment (pahud) [inline src/dual-alb-fargate-svc.ts:50]: Maybe we should expose the `protocolVersion` to the surface and allow users to specify. wdyt?

File: src/integ.default.ts [hunk 3]
```diff
@@ -223,7 +236,7 @@ const svc = new DualAlbFargateService(stack, 'Service', {
       task: orderTask,
       desiredCount: 2,
       internal: { port: 80 },
-      external: cert ? { port: 443, certificate: [cert] } : { port: 80 },
+      external: cert ? { port: 443, certificate: [cert!] } : { port: 80 },
```
Comment (pahud) [inline src/integ.default.ts:239]: `[cert]` should be enough given it's a conditional assignment. Is there any error without `!`?

**Review by pahud — CHANGES_REQUESTED** [review]
Let's create a tiny sample for gRPC maybe with golang? The python sample is way too big.

File: test/default.test.ts [hunk 4]
```diff
@@ -892,3 +894,81 @@ test('DualNlbFargateService - minimal setup both internal and external NLB', ()
...
+
+test('DualAlbFargateService - Support gRCP application', () => {
+  // GIVEN
+  // WHEN
+  const task = new ecs.FargateTaskDefinition(stack, 'pythongrcpTask', {
+    cpu: 256,
+    memoryLimitMiB: 512,
+  });
+
+  task.addContainer('python-grcp', {
```
Comment (pahud) [inline test/default.test.ts:907]:
```suggestion
  task.addContainer('go-grpc', {
```

File: test/default.test.ts [hunk 5]
```diff
@@ -892,3 +894,81 @@ test('DualNlbFargateService - minimal setup both internal and external NLB', ()
     ],
   }, ResourcePart.CompleteDefinition);
 });
+
+
+test('DualAlbFargateService - Support gRCP application', () => {
+  // GIVEN
+  // WHEN
+  const task = new ecs.FargateTaskDefinition(stack, 'pythongrcpTask', {
```
Comment (pahud) [inline test/default.test.ts:902]:
```suggestion
  const task = new ecs.FargateTaskDefinition(stack, 'grpcTask', {
```

File: test/default.test.ts [hunk 6]
```diff
@@ -892,3 +894,81 @@ test('DualNlbFargateService - minimal setup both internal and external NLB', ()
...
+        protocolVersion: elbv2.ApplicationProtocolVersion.GRPC,
+        healthCheck: {
+          healthyGrpcCodes: '12',
+        },
+      },
+    ],
+  });
+
+  // THEN
+  // We should have two NLBs
```
Comment (pahud) [inline test/default.test.ts:928]:
```suggestion
  // We should have two ALBs
```

**Review by pahud — CHANGES_REQUESTED** [review]
some minor suggested changes

File: README.md [hunk 7]
```diff
@@ -35,6 +35,21 @@ new DualAlbFargateService(stack, 'Service', {
     },
     { task: customerTask, desiredCount: 2, internal: { port: 8080 } },
     { task: productTask, desiredCount: 2, internal: { port: 9090 } },
+    
+    // support gRPC application.
+    // you can use protocolVersion to setting gRPC for your application. 
+    // you need to specify acm and grcp port.
```
Comment (pahud) [inline README.md:41]:
```suggestion
    // you need to specify acm and grpc port.
```

File: src/integ.default.ts [hunk 8]
```diff
@@ -211,6 +212,19 @@ muxTask.addContainer('mux', {
   ],
 });
 
+// go-grcp
```
Comment (pahud) [inline src/integ.default.ts:215]:
```suggestion
// go-grpc
```

File: test/default.test.ts [hunk 9]
```diff
@@ -892,3 +894,81 @@ test('DualNlbFargateService - minimal setup both internal and external NLB', ()
     ],
   }, ResourcePart.CompleteDefinition);
 });
+
+
+test('DualAlbFargateService - Support gRCP application', () => {
```
Comment (pahud) [inline test/default.test.ts:899]:
```suggestion
test('DualAlbFargateService - Support gRPC application', () => {
```

**Review by pahud — CHANGES_REQUESTED** [review]
please rename all `grcp` to `grpc`

File: src/integ.default.ts (hunk 8 above)
Comment (neilkuan) [reply src/integ.default.ts]: okay, my fault.

File: src/dual-alb-fargate-svc.ts [hunk 10]
```diff
@@ -45,7 +45,9 @@ export class DualAlbFargateService extends BaseFargateService {
       if (t.external) {
         const exttg = new elbv2.ApplicationTargetGroup(this, `${defaultContainerName}ExtTG`, {
           protocol: elbv2.ApplicationProtocol.HTTP,
+          port: t.protocolVersion === elbv2.ApplicationProtocolVersion.GRPC ? t.external!.port! : undefined,
```
Comment (pahud) [inline src/dual-alb-fargate-svc.ts:48]: what will happen if 

t.protocolVersion === elbv2.ApplicationProtocolVersion.GRPC

but t.external.port is undefined?

File: src/dual-alb-fargate-svc.ts [hunk 11]
```diff
@@ -67,7 +69,9 @@ export class DualAlbFargateService extends BaseFargateService {
       if (t.internal) {
         const inttg = new elbv2.ApplicationTargetGroup(this, `${defaultContainerName}IntTG`, {
           protocol: elbv2.ApplicationProtocol.HTTP,
+          port: t.protocolVersion === elbv2.ApplicationProtocolVersion.GRPC ? t.internal!.port! : undefined,
           vpc: this.vpc,
+          protocolVersion: t.protocolVersion ?? undefined,
```
Comment (pahud) [inline src/dual-alb-fargate-svc.ts:74]:
```suggestion
          protocolVersion: t.protocolVersion,
```

File: src/dual-alb-fargate-svc.ts [hunk 12]
```diff
@@ -45,7 +45,9 @@ export class DualAlbFargateService extends BaseFargateService {
       if (t.external) {
         const exttg = new elbv2.ApplicationTargetGroup(this, `${defaultContainerName}ExtTG`, {
           protocol: elbv2.ApplicationProtocol.HTTP,
+          port: t.protocolVersion === elbv2.ApplicationProtocolVersion.GRPC ? t.external!.port! : undefined,
           vpc: this.vpc,
+          protocolVersion: t.protocolVersion ?? undefined,
```
Comment (pahud) [inline src/dual-alb-fargate-svc.ts:50]:
```suggestion
          protocolVersion: t.protocolVersion,
```

**Review by pahud — CHANGES_REQUESTED** [review]

**Review by pahud — APPROVED** [review]

**Comment by pahud** [comment]
@all-contributors please add @neilkuan  for design and code.

**Pull request merged**